### ISSUE TITLE
[MPV Lua POSIX] Make the script work on Debian

### ISSUE THREAD

**Issue by aleksejrs** [nordlicht/nordlicht#73]
Have you tested it on Debian?

On Debian testing (future stretch), the lua-posix package is 31-3, which is too old.

I installed 33.4.0-1 from experimental, and got:

```
[mpv_nordlicht] Lua error: /usr/share/lua/5.2/posix/init.lua:29: module 'posix.ctype' not found:
[mpv_nordlicht] 	no field package.preload['posix.ctype']
[mpv_nordlicht] 	no file '/usr/local/share/lua/5.2/posix/ctype.lua'
[mpv_nordlicht] 	no file '/usr/local/share/lua/5.2/posix/ctype/init.lua'
[mpv_nordlicht] 	no file '/usr/local/lib/lua/5.2/posix/ctype.lua'
[mpv_nordlicht] 	no file '/usr/local/lib/lua/5.2/posix/ctype/init.lua'
[mpv_nordlicht] 	no file '/usr/share/lua/5.2/posix/ctype.lua'
[mpv_nordlicht] 	no file '/usr/share/lua/5.2/posix/ctype/init.lua'
[mpv_nordlicht] 	no file './posix/ctype.lua'
[mpv_nordlicht] 	no file '/etc/mpv/scripts/posix/ctype.lua'
[mpv_nordlicht] 	no file '/usr/local/lib/lua/5.2/posix/ctype.so'
[mpv_nordlicht] 	no file '/usr/lib/x86_64-linux-gnu/lua/5.2/posix/ctype.so'
[mpv_nordlicht] 	no file '/usr/lib/lua/5.2/posix/ctype.so'
[mpv_nordlicht] 	no file '/usr/local/lib/lua/5.2/loadall.so'
[mpv_nordlicht] 	no file './posix/ctype.so'
[mpv_nordlicht] 	no file '/usr/local/lib/lua/5.2/posix.so'
[mpv_nordlicht] 	no file '/usr/lib/x86_64-linux-gnu/lua/5.2/posix.so'
[mpv_nordlicht] 	no file '/usr/lib/lua/5.2/posix.so'
[mpv_nordlicht] 	no file '/usr/local/lib/lua/5.2/loadall.so'
[mpv_nordlicht] 	no file './posix.so'
```

**Comment by rohieb** [comment]
Mmh yes, this is unfortunate, we have already noticed this… Until the Debian catches up, you can try installing [lua-posix](https://github.com/luarocks/luarocks/wiki/Documentation) from [luarocks](https://github.com/luarocks/luarocks/wiki/Documentation).

**Comment by aleksejrs** [comment]
> Until the Debian catches up, you can try installing luaposix [1] from luarocks [0].

That only adds to the confusion, choosing one of the installed Lua versions, preferring 5.2 to 5.3, creating non-world-readable files.

**Comment by rohieb** [comment]
In any case, you'll need to install luaposix for the same Lua version as mpv is linked against. For the Debian package, this is currently Lua 5.2, so this looks correct. In my case, I had to install the package into `/usr/local` by using luarocks with `sudo`, but without further info what goes wrong where, I cannot help you much.

@blinry I guess we could help this situation by bundling luaposix as a git submodule under `nordlicht/utils/luaposix` and `require` it from there (it's MIT-licensed after all), and let @cinemast solve the Debian-related packaging problems ;-) Or, alternatively, deliver mpv-nordlicht as a Lua rock with dependency info to ease installation. What do you think?

**Comment by cinemast** [comment]
Is the mpv support already production ready? @blinry pointed out to me it is still in an experimental state.

**Comment by blinry** [comment]
@rohieb Yeah, bundling the libraries with nordlicht seems to be the best option we currently have. It's really frustrating that I didn't find another clean way to run processes in the background from mpv, while still being able to kill them reliably. Maybe I'll take another stab on that in the future.

@cinemast It's definitely still experimental! :)
